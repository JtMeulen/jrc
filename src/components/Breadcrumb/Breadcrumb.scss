@use '../../style/lib.scss' as *;
@use '../../style/mixins.scss';

.breadcrumb {
  @include mixins.font-family();

  display: flex;

  &__item {
    color: $jrc-color-primary;

    &:visited {
      color: $jrc-color-primary;
    }

    &:hover,
    &:focus {
      color: $jrc-color-primary-hover;
    }

    &:active {
      color: $jrc-color-primary-active;
    }

    text-decoration: none;

    &--is-active {
      text-decoration: underline;
    }
  }

  &__separator {
    margin: 0 8px;
  }

  &__separator:last-of-type {
    display: none;
  }
}
