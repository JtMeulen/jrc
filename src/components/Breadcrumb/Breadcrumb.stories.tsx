import React from "react";
import { Meta, StoryFn } from "@storybook/react";
import Breadcrumb from "./Breadcrumb";

export default {
  title: "Components/Breadcrumb",
  component: Breadcrumb,
} as Meta<typeof Breadcrumb>;

const breadcrumbData = [
  { id: "link-1", href: "/", text: "Home" },
  { id: "link-2", href: "/", text: "Menu" },
  { id: "link-3", href: "/", text: "Profile" },
];

const Template: StoryFn<typeof Breadcrumb> = (args) => <Breadcrumb {...args} />;

const defaultProps = {
  data: breadcrumbData,
};

export const Default = Template.bind({});
Default.args = { ...defaultProps };

export const ArrowSeparator = Template.bind({});
ArrowSeparator.args = { ...defaultProps, separator: "â†’" };

export const ActiveLink = Template.bind({});
ActiveLink.args = { ...defaultProps, activeId: "link-2" };
