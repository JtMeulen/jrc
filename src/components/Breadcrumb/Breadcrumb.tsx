import React from "react";
import cn from "classnames";
import "./Breadcrumb.scss";

type BreadcrumbItem = {
  id: string;
  href: string;
  text: string;
};

interface BreadcrumbProps {
  data: BreadcrumbItem[];
  separator?: string;
  activeId?: string;
}

const Breadcrumb: React.FC<BreadcrumbProps> = ({
  data = [],
  separator = "/",
  activeId,
}) => {
  return (
    <div className={cn("breadcrumb")}>
      {data.map((item) => {
        return (
          <>
            <a
              className={cn("breadcrumb__item", {
                "breadcrumb__item--is-active": activeId === item.id,
              })}
              href={item.href}
              id={item.id}
            >
              {item.text}
            </a>
            <span className="breadcrumb__separator">{separator}</span>
          </>
        );
      })}
    </div>
  );
};

export default Breadcrumb;
