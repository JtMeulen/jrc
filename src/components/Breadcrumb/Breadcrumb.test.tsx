import React from "react";
import { render } from "@testing-library/react";

import Breadcrumb from "./Breadcrumb";

describe("Breadcrumb", () => {
    // TODO: add tests for Breadcrumb
  test("renders without error", () => {
    const { container } = render(<Breadcrumb data={[]}/>);
    expect(true).toEqual(true);
  });
});
