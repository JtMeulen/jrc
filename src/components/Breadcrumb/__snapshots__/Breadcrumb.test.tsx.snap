// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb renders without error 1`] = `
<div>
  <div
    class="breadcrumb"
  >
    <a
      class="breadcrumb__item"
      href="/"
      id="link-1"
    >
      Home
    </a>
    <span
      class="breadcrumb__separator"
    >
      /
    </span>
    <a
      class="breadcrumb__item"
      href="/"
      id="link-2"
    >
      Menu
    </a>
    <span
      class="breadcrumb__separator"
    >
      /
    </span>
    <a
      class="breadcrumb__item"
      href="/"
      id="link-3"
    >
      Profile
    </a>
    <span
      class="breadcrumb__separator"
    >
      /
    </span>
  </div>
</div>
`;
